src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1272 @@
+      });
+    }
+
+    // Get notification endpoint url
+    // for QA - environment: https://service-qa.hubject.com/api/oicp/notificationmgmt/v11/charging-notifications
+    // for PROD - environment: https://service.hubject.com/api/oicp/notificationmgmt/v11/charging-notifications
+    const fullUrl = this.getEndpointUrl('notifications', ServerAction.OICP_SEND_CHARGING_NOTIFICATION_ERROR);
+
+    // Build payload
+    const payload: OICPChargingNotificationErrorCpoSend = {} as OICPChargingNotificationErrorCpoSend;
+    payload.Type = OICPChargingNotification.Error;
+    payload.SessionID = transaction.oicpData.session.id;
+    if (transaction.id) {
+      payload.CPOPartnerSessionID = String(transaction.id); // Optional
+    }
+    payload.EMPPartnerSessionID = transaction.oicpData.session.empPartnerSessionID; // Optional
+    payload.Identification = transaction.oicpData.session.identification; // Optional
+    payload.EvseID = transaction.oicpData.session.evse.EvseID;
+    payload.ErrorType = error;
+    payload.ErrorAdditionalInfo = errorAdditionalInfo; // Optional
+
+    // Log
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_SEND_CHARGING_NOTIFICATION_ERROR,
+      message: `Send Charging Notification Error for EVSE: ${payload.EvseID}`,
+      module: MODULE_NAME, method: 'sendChargingNotificationError',
+      detailedMessages: { payload }
+    });
+
+    // Call Hubject
+    await this.axiosInstance.post(fullUrl, payload
+    ).then(function(response) {
+      console.log('Success! sendChargingNotificationError: ', response);
+      notificationErrorResponse = response.data as OICPAcknowledgment;
+    }).catch((errors) => {
+      console.log('Error! sendChargingNotificationError: ', errors.response);
+    });
+    return notificationErrorResponse;
+  }
+
+  // Get ChargeBoxIds with new status notifications
+  async getChargeBoxIDsWithNewStatusNotifications(): Promise<string[]> {
+    // Get last job
+    const lastPatchJobOn = this.oicpEndpoint.lastPatchJobOn ? this.oicpEndpoint.lastPatchJobOn : new Date();
+    // Build params
+    const params = { 'dateFrom': lastPatchJobOn };
+    // Get last status notifications
+    const statusNotificationsResult = await OCPPStorage.getStatusNotifications(this.tenant.id, params, Constants.DB_PARAMS_MAX_LIMIT);
+    // Loop through notifications
+    if (statusNotificationsResult.count > 0) {
+      return statusNotificationsResult.result.map((statusNotification) => statusNotification.chargeBoxID);
+    }
+    return [];
+  }
+
+  /**
+   * Ping OICP Endpoint
+   */
+  async ping() {
+    const pingResult: any = {};
+    // Try to access base Url (GET .../versions)
+    // Access versions API
+    try {
+      // Get versions
+      const response = await this.pingEvseEndpoint();
+      // Check response
+      if (!response.Result || !(response.StatusCode.Code === OICPCode.Code000)) {
+        pingResult.statusCode = 412;
```

Comment:
Hard coded return code value. Define and use an enum.

Reply:
ok, done